packages/data-context/src/sources/ProjectDataSource.ts

```diff
@@ -101,6 +102,40 @@ export function transformSpec ({
   }
 }
 
+export function getDefaultSpecFileName (specPattern: string, fileExtensionToUse?: 'js' | 'ts') {
```

Comment:
Here are a few globs where `getDefaultSpecFileName()` doesn't return what I'd expect. I expect that there are more, that I haven't thought of. Hopefully we're considering all the patterns that the glob library we're using supports and looking at actual user glob patterns when we're writing this code and tests.

```
tests/*.js
src/*-test.js
src/*.foo.bar.js
src/prefix.*.test.js
src/*/*.test.js
src-*/**/*.test.js
src/*.test.(js|jsx)
(src|components)/**/*.test.js
```

Reply:
all these cases should be covered now - happy to add more cases if you have in mind!

https://github.com/cypress-io/cypress/pull/20654/commits/5deb317f0cf72ae584ef385345a92c43e8bf80ef#diff-59bf8023d3f96e2d91bc479be5372ebde0317cf6040e5e514ceb276ac71c1342R281-R339